### PR TITLE
Fix casing of ProblemDetails for RFC compliance

### PR DESCRIPTION
**Pull request by captainsafia**
Addresses https://github.com/dotnet/aspnetcore/issues/53639.

In .NET 8, we merged a PR to remove the custom converters used for `ProblemDetails` and `ValidationProbleemDetails` in feature of the `IgnoreWhenNull` attributes that were provided in the box by System.Text.Json.

At the same time, we removed the pre-defined type names that existed on the properties of these types.

As it turns out, this was a bad move. The RFC for problem details is particular about property keys being all lower-case ([ref](https://www.rfc-editor.org/rfc/rfc7807#section-3.1)) regardless of what serialization options the rest of the system might be using by default.

This means that are implementation is no longer RFC-compliant. Fixing this by bring backing the explicit type names.

I do plan on back porting to .NET 8.

### CODE REVIEW

**Review by halter73 — APPROVED** [review]

File: src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs [hunk 1]
```diff
@@ -54,6 +54,36 @@ public async Task WriteAsync_Works()
         Assert.Equal(expectedProblem.Instance, problemDetails.Instance);
     }
 
+    [Fact]
+    public async Task WriteAsync_Works_ProperCasing()
+    {
+        // Arrange
+        var writer = GetWriter();
```
Comment (halter73) [inline src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs:61]: For this to be a true regression test, don't we need to use a non-web-default options?

```suggestion
        var writer = GetWriter(jsonOptions: new JsonOptions
        {
            SerializerOptions = JsonSerializerOptions.Default
        });
```

File: src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs [hunk 2]
```diff
@@ -54,6 +54,36 @@ public async Task WriteAsync_Works()
...
+            ProblemDetails = expectedProblem
+        };
+
+        //Act
+        await writer.WriteAsync(problemDetailsContext);
+
+        //Assert
+        stream.Position = 0;
+        var result = await JsonSerializer.DeserializeAsync<Dictionary<string, object>>(stream, SerializerOptions);
+        Assert.Equal(result.Keys, new(new() { { "type", 0 }, { "title", 1 }, { "status", 2}, { "detail", 3 }, { "instance", 4 } }));
```
Comment (halter73) [inline src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs:84]: Should we add a test for "extensions" and "errors" (for HttpValidationProblemDetails and ValidationProblemDetails)?

File: src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs (hunk 1 above)
Comment (captainsafia) [reply src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs:61]: Good point! I missed that our "default" serialization options are the default web ones. Will fix.

File: src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs (hunk 2 above)
Comment (captainsafia) [reply src/Http/Http.Extensions/test/ProblemDetailsDefaultWriterTest.cs:84]: Can do!

**Pull request merged**